frontend/src/components/ConfirmationDialog/ConfirmationDialog.js

```diff
@@ -0,0 +1,73 @@
+import React from "react";
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle
+} from "@material-ui/core";
+import Slide from "@material-ui/core/Slide";
+import PropTypes from "prop-types";
+
+const Transition = React.forwardRef(function Transition(props, ref) {
+  return <Slide direction="down" ref={ref} {...props} />;
+});
+
+const ConfirmationDialog = ({ open, title, success, fail, children }) => {
+  const handleClose = succ => {
```

Comment:
This function is redudant. Just assign the function you need wherever you need it. For example: `<Button onClick={fail} />`